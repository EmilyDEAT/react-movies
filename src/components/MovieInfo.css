.movieInfoContainer {
  position: fixed;
  z-index: 1; 
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(35,37,42,0.7);
  overflow: scroll;
}

.movieInfo {
  background-color: rgb(35,37,42);
  color: white;
  margin: auto;
  padding: 20px;
  width: 80%;
  display: flex;
  flex-direction: column;
  max-width: 800px;
}

.hide {
  display: none;
}

.show {
  display: flex;
}

.movieInfoClose {
  width: 25px;
  align-self: flex-end;
  cursor: pointer;
  margin-bottom: 10px;
  transition: transform 300ms ease-out;
}

.movieInfoClose:hover {
  transform: scale(1.3);
}

.movieInfoImg {
  width: 100%;
}

.movieInfoTitle {
  margin-bottom: 0;
  font-size: 2em;
}

.movieInfoFacts {
  display: flex;
  flex-direction: row;
  color:  rgb(158, 161, 169);
  margin-bottom: 15px;
}

.movieInfoGenre, .movieInfoTime {
  margin-right: 0.5em;
}

.movieInfoTime::before, .movieInfoDate::before {
  content: "â—¦";
  margin-right: 0.5em;
}

.movieInfoTagline {
  font-weight: bold;
}

.movieInfoOverview {
  text-align: justify;
}