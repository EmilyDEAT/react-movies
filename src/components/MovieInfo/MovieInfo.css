.movieInfoContainer {
  position: fixed;
  z-index: 1; 
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(35,37,42,0.7);
  overflow: scroll;
}

.movieInfo {
  background-color: rgb(35,37,42);
  color: white;
  margin: auto;
  padding: 20px;
  width: 80%;
  display: flex;
  flex-direction: column;
  max-width: 800px;
}

.hide {
  display: none;
}

.show {
  display: flex;
}
.movieInfoPlaceholder {
  background-color: rgb(158, 161, 169);
  width: 100%;
  padding-top: 59.2%;
  animation: pulseDark 5s infinite;
}

@keyframes pulseDark {
  0% {
    opacity: 0.4;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.4;
  }
}

.movieInfoClose {
  width: 25px;
  align-self: flex-end;
  cursor: pointer;
  margin-bottom: 10px;
  transition: transform 300ms ease-out;
}

.movieInfoClose:hover {
  transform: scale(1.3);
}

.movieInfoImg {
  width: 100%;
}

.movieInfoImgHidden {
  display: none;
}

.movieInfoTitleContainer {
display: flex;
flex-direction: row;
align-items: baseline;
justify-content: space-between;
}

.movieInfoTitle {
  margin-bottom: 16px;
  font-size: 2em;
}

.heart {
  width: 25px;
  height: 25px;
  cursor: pointer;
}

.heart path:hover {
  fill: #EC0451;
}

.red path {
  fill: #EC0451;
}

.react-rater {
  display: flex;
  flex-direction: row;
}

.is-active{
  color: #EC0451;
}

.movieInfoVotes {
  margin-left: 0.5em;
}

.movieInfoFacts, .movieInfoRating {
  display: flex;
  flex-direction: row;
  color:  rgb(158, 161, 169);
  align-items: baseline;
}

.movieInfoFacts {
  margin-bottom: 15px;
}

.movieInfoFacts p {
  margin-top: 0;
}

.movieInfoGenre, .movieInfoTime {
  margin-right: 0.5em;
}

.movieInfoTime::before, .movieInfoDate::before {
  content: "â—¦";
  margin-right: 0.5em;
}

.movieInfoTagline {
  font-weight: bold;
}

.movieInfoOverview {
  text-align: justify;
}

.movieInfoSite {
  color: white ;
}