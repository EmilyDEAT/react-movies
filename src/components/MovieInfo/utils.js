// Time convert
const timeConvert = (n) => {
  const hours = Math.floor(n / 60);
  const minutes = n - 60 * hours;
  const prettyMinutes = minutes < 10 ? `0${minutes}` : minutes;
  return `${hours}h${prettyMinutes}`;
};

// Date convert
const dateConvert = (s) => {
  const day = s[8] + s[9];
  const month = s[5] + s[6];
  const year = s[0] + s[1] + s[2] + s[3];
  return `${day}/${month}/${year}`;
};

// Rating convert
const ratingConvert = (n) => {
  return Math.round(n / 2);
};

export { timeConvert, dateConvert, ratingConvert };
