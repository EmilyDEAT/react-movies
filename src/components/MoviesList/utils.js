// Get Movie Genre
const getGenre = (s) => {
  let idGenre = 0;
  if (s === "action") {
    idGenre = 28;
  } else if (s === "aventure") {
    idGenre = 12;
  } else if (s === "comedie") {
    idGenre = 35;
  } else if (s === "drame") {
    idGenre = 18;
  } else if (s === "familial") {
    idGenre = 10751;
  } else if (s === "fantastique") {
    idGenre = 14;
  }
  return idGenre;
};

export default getGenre;
