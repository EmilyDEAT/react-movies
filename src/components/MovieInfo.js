import React from "react";
import Rater from 'react-rater'

import cross from "../images/cross.png";

import "./MovieInfo.css";

const MovieInfo = ({ hideInfo, movie, show }) => {
  // Time convert
  const timeConvert = (n) => {
    const hours = Math.floor(n / 60);
    const minutes = n - 60 * hours;
    const prettyMinutes = minutes < 10 ? `0${minutes}` : minutes;
    return `${hours}h${prettyMinutes}`;
  };

   // Date convert
   const dateConvert = (s) => {
     const day = s[8] + s[9]
     const month = s[5] + s[6]
     const year = s[0] + s[1] + s[2] + s[3]
     return `${day}/${month}/${year}`
   }

   // Rating convert
   const ratingConvert = (n) => {
    return Math.round(n/2)
  }

  return (
    <div className={`movieInfoContainer ${show ? "show" : "hide"}`}>
      {movie === null ? (
        "Loading"
      ) : (
        <div className="movieInfo">
          <img
            className="movieInfoClose"
            src={cross}
            onClick={hideInfo}
            alt="close"
          />
          <img
            className="movieInfoImg"
            src={`https://image.tmdb.org/t/p/w500${movie.backdrop_path}`}
            alt={movie.title}
          />
          <h2 className='movieInfoTitle'>{movie.title}</h2>
          <div className='movieInfoRating'>
            <Rater total={5} rating={ratingConvert(movie.vote_average)} interactive={false} />
            <p className='movieInfoVotes'>({movie.vote_count} Votes)</p>
          </div>
          <div className='movieInfoFacts'>
            <p className="movieInfoGenre">{movie.genres[0].name}</p>
            <p className="movieInfoTime">{timeConvert(movie.runtime)}</p>
            <p className="movieInfoDate">Sortie le {dateConvert(movie.release_date)}</p>
          </div>
          <strong className="movieInfoTagline">{movie.tagline}</strong>
          <p className="movieInfoOverview">{movie.overview}</p>
          <a className="movieInfoSite" href={movie.homepage} target="_blank" rel="noopener noreferrer">Plus d'infos</a>
        </div>
      )}
    </div>
  );
};

export default MovieInfo;
