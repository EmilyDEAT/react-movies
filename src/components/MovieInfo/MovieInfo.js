import React, { useState, useEffect } from "react";
import Rater from "react-rater";

import { timeConvert, dateConvert, ratingConvert } from "./utils";

import { ReactComponent as Heart } from "../../images/heart.svg";
import cross from "../../images/cross.png";

import "./MovieInfo.css";

const MovieInfo = ({ hideInfo, movie, show, add, isFavorite }) => {
  const [loaded, setLoaded] = useState(false);

  const handleImgLoaded = () => {
    setLoaded(true);
  };

  useEffect(() => {
    setLoaded(false);
  }, [movie]);

  return (
    <div
      className={`movieInfoContainer ${show ? "show" : "hide"}`}
      onClick={hideInfo}
    >
      {movie === null ? (
        <div className="loader"></div>
      ) : (
        <div className="movieInfo" onClick={(e) => e.stopPropagation()}>
          <img
            className="movieInfoClose"
            src={cross}
            onClick={hideInfo}
            alt="close"
          />
          {loaded ? null : <div className="movieInfoPlaceholder"></div>}
          <img
            className={loaded ? "movieInfoImg" : "movieInfoImgHidden"}
            src={`https://image.tmdb.org/t/p/w500${movie.backdrop_path}`}
            alt={movie.title}
            onLoad={handleImgLoaded}
          />
          <div className="movieInfoTitleContainer">
            <h2 className="movieInfoTitle">{movie.title}</h2>
            <Heart
              className={`heart ${isFavorite ? "red" : ""}`}
              onClick={add}
            />
          </div>
          <div className="movieInfoRating">
            <Rater
              total={5}
              rating={ratingConvert(movie.vote_average)}
              interactive={false}
            />
            <p className="movieInfoVotes">({movie.vote_count} Votes)</p>
          </div>
          <div className="movieInfoFacts">
            <p className="movieInfoGenre">{movie.genres[0].name}</p>
            <p className="movieInfoTime">{timeConvert(movie.runtime)}</p>
            <p className="movieInfoDate">
              Sortie le {dateConvert(movie.release_date)}
            </p>
          </div>
          <strong className="movieInfoTagline">{movie.tagline}</strong>
          <p className="movieInfoOverview">{movie.overview}</p>
          <a
            className="movieInfoSite"
            href={movie.homepage}
            target="_blank"
            rel="noopener noreferrer"
          >
            Plus d'infos
          </a>
        </div>
      )}
    </div>
  );
};

export default MovieInfo;
