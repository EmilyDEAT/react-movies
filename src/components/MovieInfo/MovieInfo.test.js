import React from "react";
import renderer from "react-test-renderer";

import MovieInfo from "./MovieInfo";
import { timeConvert, dateConvert, ratingConvert } from "./utils";

it("MovieInfo renders correctly", () => {
  const props = {
    hideInfo: () => {},
    movie: {
      backdrop_path: "",
      title: "",
      vote_average: 1,
      vote_count: 1,
      genres: [{ name: "" }],
      runtime: 100,
      release_date: "",
      tagline: "",
      overview: "",
      homepage: "",
    },
    show: () => {},
    add: () => {},
    isFavorite: true,
  };
  const movieInfo = renderer.create(<MovieInfo {...props} />).toJSON();
  expect(movieInfo).toMatchSnapshot();
});

it("timeConvert returns time correctly", () => {
  expect(timeConvert(123)).toEqual("2h03");
  expect(timeConvert(130)).toEqual("2h10");
  expect(timeConvert(50)).toEqual("0h50");
});

it("dateConvert returns date correctly", () => {
  expect(dateConvert("2019-11-20")).toEqual("20/11/2019");
});

it("ratingConvert returns rating correctly", () => {
  expect(ratingConvert(8.2)).toEqual(4);
  expect(ratingConvert(7.2)).toEqual(4);
});
