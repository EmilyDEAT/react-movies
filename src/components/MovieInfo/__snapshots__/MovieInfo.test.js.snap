// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo renders correctly 1`] = `
<div
  className="movieInfoContainer show"
  onClick={[Function]}
>
  <div
    className="movieInfo"
    onClick={[Function]}
  >
    <img
      alt="close"
      className="movieInfoClose"
      onClick={[Function]}
      src="cross.png"
    />
    <div
      className="movieInfoPlaceholder"
    />
    <img
      alt=""
      className="movieInfoImgHidden"
      onLoad={[Function]}
      src="https://image.tmdb.org/t/p/w500"
    />
    <div
      className="movieInfoTitleContainer"
    >
      <h2
        className="movieInfoTitle"
      >
        
      </h2>
      <svg
        className="heart red"
        onClick={[Function]}
      >
        heart.svg
      </svg>
    </div>
    <div
      className="movieInfoRating"
    >
      <div
        className="react-rater"
      >
        <div
          onClick={null}
          onMouseOver={null}
        >
          <div
            className="react-rater-star is-disabled is-active"
          >
            ★
          </div>
        </div>
        <div
          onClick={null}
          onMouseOver={null}
        >
          <div
            className="react-rater-star is-disabled"
          >
            ★
          </div>
        </div>
        <div
          onClick={null}
          onMouseOver={null}
        >
          <div
            className="react-rater-star is-disabled"
          >
            ★
          </div>
        </div>
        <div
          onClick={null}
          onMouseOver={null}
        >
          <div
            className="react-rater-star is-disabled"
          >
            ★
          </div>
        </div>
        <div
          onClick={null}
          onMouseOver={null}
        >
          <div
            className="react-rater-star is-disabled"
          >
            ★
          </div>
        </div>
      </div>
      <p
        className="movieInfoVotes"
      >
        (
        1
         Votes)
      </p>
    </div>
    <div
      className="movieInfoFacts"
    >
      <p
        className="movieInfoGenre"
      >
        
      </p>
      <p
        className="movieInfoTime"
      >
        1h40
      </p>
      <p
        className="movieInfoDate"
      >
        Sortie le 
        NaN/NaN/NaN
      </p>
    </div>
    <strong
      className="movieInfoTagline"
    >
      
    </strong>
    <p
      className="movieInfoOverview"
    >
      
    </p>
    <a
      className="movieInfoSite"
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      Plus d'infos
    </a>
  </div>
</div>
`;
